pkg/apis/machine/types.go

```diff
@@ -213,6 +213,24 @@ const (
 	MachineStateSuccessful MachineState = "Successful"
 )
 
+// MachineLastOperationType is a label for the last operation performed on a machine object.
+type MachineLastOperationType string
+
+// These are the valid statuses of machines.
+const (
+	// MachineLastOperationCreate indicates that the last operation was a create
+	MachineLastOperationCreate MachineLastOperationType = "Create"
+
+	// MachineLastOperationUpdate indicates that the last operation was an update
+	MachineLastOperationUpdate MachineLastOperationType = "Update"
+
+	// MachineLastOperationHealthCheck indicates that the last operation was a create
+	MachineLastOperationHealthCheck MachineLastOperationType = "HealthCheck"
+
+	// MachineLastOperationDelete indicates that the last operation was a create
+	MachineLastOperationDelete MachineLastOperationType = "Delete"
+)
```

Comment:
I think we should rename these to something more generic -Eg. `MachineOperationCreate`
Multiple structs will be able to use it in that case, and not just `LastOperation`

Reply:
Isn't HealthCheck is something we would do[if not now] periodically anyways, and should not be part of the MachineLastOperationType struct? Just checking if semanticcs really match.

Reply:
@ggaurav10 I don't think there are any other structs making use of the key word create. Correct me if I am wrong.

@hardikdr - Yes, its true that HealthCheck operation is done periodically. But Here we are updating the lastOperation only when the machine state is going to unknown, I thought it would make sense to keep in the last operation status as it would give us an idea when an machine just went unhealthy. And moreover lastOperation refers to the last operation taken by the controller right? and in this case controller updated the health status. And I thought this is something that the object must depict?

Reply:
I meant from future extensibility point of view. This `type` is made generic, it can be used elsewhere also.

Reply:
As suggested by @ggaurav10. I have updated `MachineLastOperation` variables to `MachineOperation` variables.